### PR TITLE
Parsing of tiller request details accepts an AppRepositoryResourceName

### PR DESCRIPTION
**Pull request by absoludity**
Initial diff related to #1110 enabling Tiller to parse requests containing the name of an app repository custom resource. See that issue for more details.

### CODE REVIEW

**Comment by absoludity** [comment]
Seems to be failing CI because we're doing the checkout in GOPATH and so go modules are not being used. I see there's a ticket to update the README, but who has access to the circle ci setup? @andresmgot or @prydonius ?

File: pkg/chart/chart.go [hunk 1]
```diff
@@ -61,6 +61,11 @@ func init() {
 type Details struct {
 	// RepoURL is the URL of the repository. Defaults to stable repo.
 	RepoURL string `json:"repoUrl,omitempty"`
+	// AppRepositoryResourceURL specifies an app repository resource to use
+	// for the request.
```
Comment (andresmgot) [inline pkg/chart/chart.go]: I think you changed the property but not this comment.

**Comment by andresmgot** [comment]
> Seems to be failing CI because we're doing the checkout in GOPATH and so go modules are not being used. I see there's a ticket to update the README, but who has access to the circle ci setup? 

We are using the vendored packages on purpose (at least for the moment). You can update it running `go mod vendor`.

I am not sure why you mention the circle ci setup, what do you need from there? what's the ticket you mention? Also, in theory you should be able to access/configure CircleCI as a member, you just need to sign-up and follow the project. Let me know if that is not the case.

**Comment by absoludity** [comment]
> > Seems to be failing CI because we're doing the checkout in GOPATH and so go modules are not being used. I see there's a ticket to update the README, but who has access to the circle ci setup?
> 
> We are using the vendored packages on purpose (at least for the moment). You can update it running `go mod vendor`.
> 
> I am not sure why you mention the circle ci setup, what do you need from there?

I just saw that one of the ci/circleci tests had failed as I was leaving, quickly checked and saw it was due to the new mod (and forgot that we still vendor). Thanks, updated.

> what's the ticket you mention?

It was actually #774 that I meant about updating the README to not specify GOPATH, but checking it is actually unrelated to go mod. If it's not already, I'll update it tomorrow to reflect what it should be now (and fix it).

> Also, in theory you should be able to access/configure CircleCI as a member, you just need to sign-up and follow the project. Let me know if that is not the case.

Should be - I'll check tomorrow. Thanks Andres.

**Pull request merged**